concepts/error-handling/about.md

```diff
@@ -1 +1,113 @@
 # Error Handling
+
+Cairo provides `Error Handling` techniques to address issues, and manage errors. It handles the enexpected behaviour to run the code smoothly while execution.
+For example *devide by zero* situation.
+> `Err("Cannot divide by zero")`
+
+There are multiple ways to hendle errors in `Cairo`
+- Recoverable Errors
+    - Result enum: Handles potential errors in function.
+    - Using `?`: Operator for error propagation.
+- Unrecoverable errors
+    - `panic`: Used to handle unrecoverable errors. It will stop the execution and print error.
+    - `nopanic`: Used to avoid panic and handle error whithout stopping it.
+
+## The Result Enum
+This approach handles errors gracefully, providing feedback insted of crashing or producing undefined behaviiour.
+
+**The code snippet below shows how enums could be defined:**
+```
+enum Result<T, E> {
+    Ok: T,
+    Err: E,
+}
+```
+
+> Let's Take an example of `Devide By Zero` 
+```
+fn divide(a: felt, b: felt) -> Result<felt, &'static str> {
+    if b == 0 {
+        return Err("Error: Division by zero"); // Return an error if division by zero is attempted
+    }
+    Ok(a / b)
+}
+```
+
+```
+// Test with division by zero
+match divide(10, 0) {
+    Ok(result) => println!("Division successful: {}", result),
+    Err(e) => println!("{}", e),
+}
+```
+## The `?` Operator
+`?` operator is used for minimilistic(concise) errors. Allow to write clean and readable code without explicit values.
+
+>Example
+```
+fn safe_divide(a: felt, b: felt) -> Result<felt, &'static str> {
+    // The ? operator automatically generate the error if it occurs
+    let result = divide(a, b)?;
+    Ok(result)
+}
+```
+
+```
+match safe_divide(10, 0) {
+        Ok(value) => println!("Division successful: {}", value),
+        Err(e) => println!("{}", e),
+    }
+```
+
+## Error with `panic!` 
+When program encounters an error which is unrecoverable then a `panic!` occures. This will stop execution and helpful for the case where continuing execution doesn't make any sense.
+
+> Example
+```
+fn main() {
```

Comment:
Use this example to show the use of a plain `panic`, and move this example to the end of the section so that students still see the macro as well. Add a brief sentence explaining the differences between the two.